src/test/hosting/cloudRunProxy.spec.ts

```diff
@@ -0,0 +1,288 @@
+import { expect } from "chai";
+import * as express from "express";
+import * as nock from "nock";
+import * as sinon from "sinon";
+import * as supertest from "supertest";
+
+import { cloudRunApiOrigin } from "../../api";
+import cloudRunProxy, {
+  CloudRunProxyOptions,
+  CloudRunProxyRewrite,
+} from "../../hosting/cloudRunProxy";
+
+describe("cloudRunProxy", () => {
+  const fakeOptions: CloudRunProxyOptions = {
+    project: "project-foo",
+  };
+
+  const fakeRewrite: CloudRunProxyRewrite = { run: { serviceId: "helloworld" } };
+
+  const cloudRunServiceOrigin = "https://helloworld-hash-uc.a.run.app";
+  const cloudRunServiceOriginAsia = "https://helloworld-hash-as.a.run.app";
+
+  before(() => {
```

Comment:
an overall test style thing: the stub for each of the tests should be with that test. otherwise, it's really difficult to understand _which_ mocked API call you're supposed to be hitting. You could be hitting _more than one_ in your test case and not know it.

Change the `after` to an `afterEach` and move the `nock` calls to the appropriate functions. I don't think that should have more side effects...

Reply:
Yeah I noticed that when i started to get into the more advanced stuff (see the `.persist()` for the caching test).  I left them that way just to conform with the way that Functions was doing it.  But you are right, the nocks can only be used once, and it would be a bigger pain later when someone was adding to this.

Reply:
Alright, Nocks have been moved.  I'm planning on moving the ones in the Functions proxy as well, just so everything is consistent.  However I'm going to wait and do a separate PR for that (since it is not related to this change at all).